users/models.py

```diff
@@ -78,6 +99,37 @@
             self.avatar = self.process_avatar(self.avatar)
         super().save(*args, **kwargs)
 
+    def has_permission(self, post_id: int, permission_name: str) -> bool:
+        Post = apps.get_model("core", "Post")
+        CommunityMember = apps.get_model("core", "CommunityMember")
+
+        try:
+            post = Post.objects.get(id=post_id)
+        except Post.DoesNotExist:
+            print(f"Post with id {post_id} does not exist.")
+            return False
+
+        match permission_name:
```

Comment:
@mateuszfelczak AI suggestion here is pretty good but I would do it bit different way.
Instead of using lambda, move the code from line 111 to 120 into a function, called, eg.
__check_permission_post_edit (two underscores for private method)
Then do  (pseudocode)
```
        permission_checkers = {
            "edit": __check_permission_post_edit(PARAMETERS)
            # Add more permissions here as needed
        }
```
this way we have really clean has_permission method in models and __check_permission_post_edit is separate function. This again, gives us the option to write tests only for __check_permission_post_edit, not for the whole has_permssion thing. So please also rewrite your already good test into much better ones
- all the tests you have for has_permission can be done on __check_permission_post_edit
- please add two tests (positive and negative) for user.has_has_permission with the edit permission. This willl be integration tests showing that __check_permission_post_edit is actually used. 
-- Or you can get really fancy and try to use spy while calling has_permission to show that __check_permission_post_edit was called. Code example below.


==== Using mock (spy) to show function was called ===

```python
from unittest.mock import MagicMock

def some_function():
    pass

def function_to_test():
    some_function()

def test_function_to_test():
    mock = MagicMock()
    globals()['some_function'] = mock
    function_to_test()
    mock.assert_called_once()

if __name__ == "__main__":
    import pytest
    pytest.main()
```

In this example:

1. Create a `MagicMock` for `some_function`.
2. Replace the original `some_function` with the mock.
3. Call `function_to_test`.
4. Verify that `some_function` was called exactly once using `assert_called_once`.

This is a simple and effective way to verify function calls in your tests. Happy testing!